bin/apisix

```diff
@@ -1,36 +1,40 @@
-#!/usr/bin/env lua
+#!/bin/bash
 
---
--- Licensed to the Apache Software Foundation (ASF) under one or more
--- contributor license agreements.  See the NOTICE file distributed with
--- this work for additional information regarding copyright ownership.
--- The ASF licenses this file to You under the Apache License, Version 2.0
--- (the "License"); you may not use this file except in compliance with
--- the License.  You may obtain a copy of the License at
---
---     http://www.apache.org/licenses/LICENSE-2.0
---
--- Unless required by applicable law or agreed to in writing, software
--- distributed under the License is distributed on an "AS IS" BASIS,
--- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
--- See the License for the specific language governing permissions and
--- limitations under the License.
---
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
-local pkg_cpath_org = package.cpath
-local pkg_path_org = package.path
+if [ -s './apisix/cli/apisix.lua' ]; then
+    APISIX_LUA=./apisix/cli/apisix.lua
+elif [ -s '/usr/local/share/lua/5.1/apisix/cli/apisix.lua' ]; then
+    APISIX_LUA=/usr/local/share/lua/5.1/apisix/cli/apisix.lua
+else
+    APISIX_LUA=/usr/local/apisix/apisix/cli/apisix.lua
+fi
 
-local apisix_home = "/usr/local/apisix"
-local pkg_cpath = apisix_home .. "/deps/lib64/lua/5.1/?.so;"
-                  .. apisix_home .. "/deps/lib/lua/5.1/?.so;;"
-local pkg_path = apisix_home .. "/deps/share/lua/5.1/?.lua;;"
+# find the openresty
+OR_BIN=$(which openresty)
+OR_EXEC=${OR_BIN:-'/usr/local/openresty-debug/bin/openresty'}
 
--- modify the load path to load our dependencies
-package.cpath = pkg_cpath .. pkg_cpath_org
-package.path  = pkg_path .. pkg_path_org
+if [[ -e $OR_EXEC ]]; then
+    # find the luajit binary of openresty
+    LUAJIT_BIN=$(${OR_EXEC} -V 2>&1 | grep prefix | grep -Eo 'prefix=(.*)/nginx\s+--' | grep -Eo '/.*/')luajit/bin/luajit
 
--- pass path to construct the final result
-local env = require("apisix.cli.env")(apisix_home, pkg_cpath_org, pkg_path_org)
-local ops = require("apisix.cli.ops")
-
-ops.execute(env, arg)
+    # use the luajit of openresty
```

Comment:
Why not:

```shell
$LUAJIT_BIN -e "require (\"apisix.cli.apisix\")"
```

As i said before, once you installed APISIX correctly, the code base can be located in one of the lua path, we don't really care about the absolute path you wrote above (and it's not so portable, may fail on some OS).

Reply:
The absolute path is not portable really. 
The installed path of APISIX may not be the lua path, it is a troublesome to query lua path in bash. 
I've ever tried `$LUAJIT_BIN -e "require (\"apisix.cli.apisix\")"`, but it could not work well in shell script, the argument always error.

Maybe it is better to change `bin/apisix` to lua script？After all, it's easier to get package.path from lua script.

Reply:
Have you tried the way that i mentioned above?

Reply:
I've already tried `$LUAJIT_BIN -e "require (\"apisix.cli.apisix\")"`, it could not work well:
1. In the development(APISIX not installed), the deps will install in local dir, when run './bin/apisix init', it will find deps in luajit default path not local, therefore, the deps should be solved: `mkdir -p /usr/local/apisix/deps/ && cp -r deps/* /usr/local/apisix/deps/`  or modify `make deps` in Makefile.
2. The parameter error when run `apisix init` or `./bin/apisix init`:
`/usr/local/openresty/luajit/bin/luajit: cannot open init: No such file or directory`

How to deal with parameter `init`, could you help me? Thanks~